charter.md

```diff
@@ -2,24 +2,45 @@
+Recognizing the need for a standardized security protocol to support group key establishment, authentication, and confidentiality services for messaging, the IETF has specified the Messaging Layer Security (MLS) protocol [I-D.ietf-mls-protocol] and architecture [I-D.ietf-mls-architecture]. MLS is agnostic to the identity system used within any given messaging service; it provides confidentiality of sessions once the participants in a conversation have been identified. To achieve interoperable messaging, the MIMI working group will identify profiles of one or more existing identity systems suitable for establishing end-to-end cryptographic identity across messaging services, assuming the use of MLS for key establishment. 
 
-Achieving interoperability requires support for user discovery, i.e., the ability for a sending user in one application to find a recipient user in a different application. The working group will specify one or more mechanisms to enable user discovery, together with best practice recommendations for functionality, configuration options, and other aspects that may require implementation outside the messaging applications themselves (e.g., in operating systems). At a minimum, user discovery based on phone numbers and email addresses must both be specified. New means of user discovery must support common user control and privacy features (e.g., allowing users to block contacts, control their own discoverability, etc.). Express user preferences about discoverability must be respected. User discovery mechanisms should also be designed to minimize the accumulation of user contact or relationship data by intermediaries.
+Achieving interoperability requires support for user discovery, i.e., the ability for a sending user in one application to find a recipient user in a different application. The working group will specify one or more mechanisms to enable user discovery, together with best practice recommendations for functionality, configuration options, and other aspects. At a minimum, user discovery within a single service, based on Internet domain-based addresses (ex: email addresses), and phone numbers should both be specified. New means of user discovery must support common user control and privacy features (e.g., allowing users to block contacts, control their own discoverability, etc.). Express user preferences about discoverability must be respected. User discovery mechanisms should also be designed to minimize the accumulation of user contact or relationship data by intermediaries.
```

Comment:
I don't get the "within a single service" bit. Each individual service already has its own user discovery mechanism, I thought the point was to support discovery of users across services.

Reply:
I think the point we discussed is that discovering which services someone is using is an unsolvable tarball if you care about spam and privacy. The individual services have their own proprietary user discovery mechanisms which nobody else outside can use. That is what we need to standardize.

Reply:
At the end of the day, I think the functional requirement here is that a sender using App A needs to be able to type in an identifier for a recipient and using that information, App A needs to be able to determine if it can connect the sender to the recipient. For charter scoping purposes I’m not even sure that we should assume the sender will specify the recipient’s app. Therefore, at a minimum I think phone number and email address look-ups — as an overlay to internal app discovery mechanisms — need to be supported.